src/cryptoadvance/specter/templates/includes/tx-data.html

```diff
@@ -67,7 +67,26 @@ <h2>Transaction details</h2><br>
                 let walletName = jsonResponse.walletName;
                 let tx = jsonResponse.tx;
                 let walletLink = `{{ url_for('wallets_endpoint.wallet', wallet_alias='WALLET_ALIAS') }}`.replace("WALLET_ALIAS", self.wallet);
-                let rawtxHTML = `
+                let rawtxHTML = "";
+                if(tx.is_purged) {
+                    rawtxHTML += `
+                    <div class="warning">
+                        <p><b>This transaction (aka "tx") could not be found in your node's mempool!</b></p>
+                        <p style="text-align:left;">If the bitcoin network is very busy, nodes will start purging the pending txs with the lowest fees. Txs older than two weeks are also purged.</p>
+                        <p style="text-align:left;">A purged tx will never complete. Abandoning this tx will clear it from your wallet, making those funds spendable once again.</p>
+                        <p style="text-align:left;">But first verify that other nodes have also purged your tx! Enter the tx id into a public block explorer (warning: slight privacy leak). If the tx cannot be found, it is safe to abandon this tx.</p>
+
+                        <br/><br/>
+                        <form method="POST" action="${walletLink}history">
+                            <input type="hidden" class="csrf-token" name="csrf_token" value="${csrf_token}"/>
```

Comment:
Doesn't really matter, but are you sure it's not accessible simply via `"{{ csrf_token() }}"` and have to be added to the `wallet_history.jinja`?

Reply:
Ah, I should have tried the obvious. I had assumed the naming convention implied that the *.html were being handled as static (as opposed to the *.jinja templates) but didn't notice that there were of course template injections happening in there already.